src/Lasso.jl

```diff
@@ -209,6 +209,10 @@ const MAX_DEV_FRAC = 0.999
 # Compute automatic λ values based on λmax and λminratio
 function computeλ(λmax, λminratio, α, nλ)
     λmax /= α
+    if isapprox(λmax, 0; atol=1e-10)  # then assuming λmax = 0
```

Comment:
This is tricky because I think lambda is not unitless, so if it is small or not depends on the data given.
How does glmnet in R (or GLMNet.jl) handle this case?

Reply:
The reason I've changed the equality check to an isapprox() check is due to floating point arithmetic leading to a lambdamax that should actually be zero being very close to zero but non-zero instead. Simple example when this happens is a design matrix X with entries sampled from U[0, 1] and y a non-zero vector with identical entries. 

I agree, the data could be such that lambdamax is genuinely very small but non-zero.

That said, I think it would be very rare to encounter such data in practice... especially since lambdamax (for the linear model) scales linearly with X and y, and we tend to standardise these. 

I see two approaches going forward:

1) Keep this check as is - the case where it would fail to produce correct output basically never occurs anyway. 

2) Revert back to the equality check. The real case in which the package failed was the case where lambdamax was _exactly_ zero, anyway. Moreover, even if lambdamax should be zero but instead is a very small number, there is no major problem: the solver works very fast and it is clear from the output that every value of lambda yields zero active coefficients. 

I'll leave it to you to decide 😃.

Reply:
Additionally: I'm not sure how glmnet in R or Julia handles this.